sdk/nodejs/iot/topicRule.ts

```diff
@@ -116,20 +116,20 @@ export class TopicRule extends pulumi.CustomResource {
      * Configuration block with error action to be associated with the rule. See the documentation for `cloudwatchAlarm`, `cloudwatchLogs`, `cloudwatchMetric`, `dynamodb`, `dynamodbv2`, `elasticsearch`, `firehose`, `http`, `iotAnalytics`, `iotEvents`, `kafka`, `kinesis`, `lambda`, `republish`, `s3`, `sns`, `sqs`, `stepFunctions`, `timestream` configuration blocks for further configuration details.
      */
     public readonly errorAction!: pulumi.Output<outputs.iot.TopicRuleErrorAction | undefined>;
-    public readonly firehose!: pulumi.Output<outputs.iot.TopicRuleFirehose | undefined>;
+    public readonly firehoses!: pulumi.Output<outputs.iot.TopicRuleFirehose[] | undefined>;
     public readonly https!: pulumi.Output<outputs.iot.TopicRuleHttp[] | undefined>;
     public readonly iotAnalytics!: pulumi.Output<outputs.iot.TopicRuleIotAnalytic[] | undefined>;
     public readonly iotEvents!: pulumi.Output<outputs.iot.TopicRuleIotEvent[] | undefined>;
     public readonly kafkas!: pulumi.Output<outputs.iot.TopicRuleKafka[] | undefined>;
-    public readonly kinesis!: pulumi.Output<outputs.iot.TopicRuleKinesis | undefined>;
-    public readonly lambda!: pulumi.Output<outputs.iot.TopicRuleLambda | undefined>;
+    public readonly kineses!: pulumi.Output<outputs.iot.TopicRuleKinese[] | undefined>;
+    public readonly lambdas!: pulumi.Output<outputs.iot.TopicRuleLambda[] | undefined>;
     /**
      * The name of the rule.
      */
     public readonly name!: pulumi.Output<string>;
-    public readonly republish!: pulumi.Output<outputs.iot.TopicRuleRepublish | undefined>;
-    public readonly s3!: pulumi.Output<outputs.iot.TopicRuleS3 | undefined>;
-    public readonly sns!: pulumi.Output<outputs.iot.TopicRuleSns | undefined>;
+    public readonly republishes!: pulumi.Output<outputs.iot.TopicRuleRepublish[] | undefined>;
+    public readonly s3s!: pulumi.Output<outputs.iot.TopicRuleS3[] | undefined>;
+    public readonly sns!: pulumi.Output<outputs.iot.TopicRuleSn[] | undefined>;
```

Comment:
Some of the pluralization/depluralization is really weird here.

Reply:
`s3s` is fixed by https://github.com/pulumi/pulumi-aws/pull/2615/commits/399d2b5866310ea5afdf4e34835cc33c5add045e.